lib/screens/qr_code_screen.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:barcode_widget/barcode_widget.dart';
+
+class QrCodeScreen extends StatefulWidget {
+  final String data;
+  final String deviceId;
+  final int time;
+  const QrCodeScreen({Key? key, required this.data, required this.deviceId, required this.time}) : super(key: key);
+
+  @override
+  _QrCodeState createState() => _QrCodeState();
+}
+
+class _QrCodeState extends State<QrCodeScreen> {
+  int startTime = 0;
+  final DateTime _now = DateTime.now();
+  late Timer timer;
+  // starts the timer when seed is generated and displayed
+  void startTimer() {
+    int convertedtime = (_now.millisecondsSinceEpoch / 1000).floor();
+    int start = convertedtime == widget.time ? 15 : convertedtime - widget.time;
+    startTime = start;
+    const oneSec = Duration(seconds: 1);
+    timer = Timer.periodic(
+      oneSec,
+      (Timer timer) {
+        if (startTime == 0) {
+          setState(() {
+            timer.cancel();                 
+          });
+        } else {
+          setState(() {
+            startTime--;
+          });
+        }
+      },
+    );
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    startTimer();
+  }
+
+  @override
+  void dispose() {
+    timer.cancel();
```

Comment:
**suggestion:** it's best practice to account for if the timer has already been disposed
```suggestion
    timer?.cancel();
```